dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -4203,35 +4204,48 @@ else if( field.getFieldType().equals(Field.FieldType.CATEGORY.toString()) ) {
 
                     buffy.append(" +(live:true working:true)");
                     buffy.append(" +structureInode:" + contentlet.getStructureInode());
-                    buffy.append(" +languageId:" + contentlet.getLanguageId());
                     buffy.append(" +(working:true live:true)");
-                    if(UtilMethods.isSet(contentlet.getIdentifier())){
+                    if (UtilMethods.isSet(contentlet.getIdentifier())) {
                         buffy.append(" -(identifier:" + contentlet.getIdentifier() + ")");
+                        buffy.append(" +languageId:" + contentlet.getLanguageId());
                     }
-                    buffy.append(" +" + contentlet.getStructure().getVelocityVarName() + "." + field.getVelocityVarName() + ":\"" + escape(getFieldValue(contentlet, field).toString()) + "\"");
+                    buffy.append(" +" + contentlet.getStructure().getVelocityVarName() + "." + field
+                            .getVelocityVarName() + ":");
+                    buffy.append(
+                            (field.getDataType().contains(DataTypes.INTEGER.toString()) || field
+                                    .getDataType().contains(DataTypes.FLOAT.toString())) ? escape(
+                                    getFieldValue(contentlet, field).toString())
+                                    : "\"" + escape(getFieldValue(contentlet, field).toString())
+                                            + "\"");
                     List<ContentletSearch> contentlets = new ArrayList<ContentletSearch>();
                     try {
-                        contentlets = searchIndex(buffy.toString(), -1, 0, "inode", APILocator.getUserAPI().getSystemUser(), false);
+                        contentlets = searchIndex(buffy.toString(), -1, 0, "inode",
+                                APILocator.getUserAPI().getSystemUser(), false);
                     } catch (Exception e) {
-                        Logger.error(this, e.getMessage(),e);
-                        throw new DotContentletValidationException(e.getMessage(),e);
+                        Logger.error(this, e.getMessage(), e);
+                        throw new DotContentletValidationException(e.getMessage(), e);
                     }
                     int size = contentlets.size();
-                    if(size > 0 && !hasError){
+                    if (size > 0 && !hasError) {
 
                         Boolean unique = true;
                         for (ContentletSearch contentletSearch : contentlets) {
                             Contentlet c = conFac.find(contentletSearch.getInode());
                             Map<String, Object> cMap = c.getMap();
                             Object obj = cMap.get(field.getVelocityVarName());
 
-                            if(((String) obj).equalsIgnoreCase(((String) o))) {
+                            if ((field.getDataType().contains(DataTypes.INTEGER.toString()) || field
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this conditional structure or edit its code blocks so that they're not all the same. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3923)